src/components/player-info.js

```diff
@@ -1,5 +1,28 @@
 import { injectCustomShaderChunks } from "../utils/media-utils";
 import { AVATAR_TYPES } from "../utils/avatar-utils";
+
+function ensureAvatarNodes(json) {
+  const { nodes } = json;
+  if (!nodes.some(node => node.name === "Head")) {
+    // If the avatar model doesn't have a Head node. The user has probably chosen a custom GLB.
+    // So, we need to construct a suitable hierarchy for avatar functionality to work.
+    // We re-parent the original root node to the Head node and set the scene root to a new AvatarRoot.
+    nodes.push({ name: "LeftEye", extensions: { MOZ_hubs_components: {} } });
+    nodes.push({ name: "RightEye", extensions: { MOZ_hubs_components: {} } });
+    nodes.push({
+      name: "Head",
+      children: [0, nodes.length - 1, nodes.length - 2],
+      extensions: { MOZ_hubs_components: { "scale-audio-feedback": "" } }
+    });
+    nodes.push({ name: "Neck", children: [nodes.length - 1] });
+    nodes.push({ name: "Spine", children: [nodes.length - 1] });
+    nodes.push({ name: "Hips", children: [nodes.length - 1] });
+    nodes.push({ name: "AvatarRoot", children: [nodes.length - 1] });
+    json.scenes[0].nodes[0] = nodes.length - 1;
```

Comment:
We should use the `json.scene`th scene if its defined, otherwise default to 0th.